evaluator.js

```diff
@@ -622,16 +620,26 @@ function is_empty_environment(env) {
    return is_null(env);
 }
 
+// set_value is used for setting a given value
+// at the head of given list, as well as to
+// store a boolean indicating whether the name of
+// the given value was declared as a constant or not
+
+function set_value(vals, val, is_constant) {
```

Comment:
Can we place `set_value` function definition inside the `set_name_value` function definition? Since it's not used elsewhere

Reply:
Yeah that's a good point. Although I think it will hurt readability, what about you?

Reply:
Either way works for me. We should check if this type of stuff (coding style / convention) affects our final evaluation.

Reply:
Actually, on second thought, let's place `set_value` inside `set_name_value`. 
It will be in line with the existing style of having the function definitions of helper functions inside the functions where they are used?

Reply:
Alright.